docs/sources/examples/getting_started_with_compose_and_osclass.md

```diff
@@ -0,0 +1,90 @@
+page_title: Getting started with compose and Osclass
```

Comment:
Very nice contribution @XaviOutside.  Some of the structure in your documentation is non-standard and uses non-English phrasing.  So...

---> https://gist.github.com/moxiegirl/86ef1eb7320bf03133ac

page_title: Getting started with Docker Compose and Osclass
page_description: Migrating and running an Osclass installation.
page_keywords: docker, example, package installation, osclass, compose

# Getting started with Docker Compose and Osclass

This document teaches you how to migrate an Osclass installation to a Docker
infrastructure. When you are done, your Docker infrastructure will contain six Docker containers, three  
server containers and three volume containers:

* Mysql database
* Postfix server
* Apache2 Webserver
* Osclass volume 
* Database volume
* Backup volume

This configuration was tested on Osclass 3.5.

## Prerequisites

Before you begin, make sure you have installed the following software:

* VirtualBox
* Docker
* Docker Machine 
* Docker compose 

Refer to the appropriated Docker documentation for different distributions for installing.

## Set up the servers

1. Create the virtualbox machine.

        $ docker-machine create --driver virtualbox dockerenv 

        $ docker-machine active dockerenv

        $ $(docker-machine env dockerenv)

2. Download the scripts using `git clone`.

        $ git clone https://github.com/XaviOutside/MIGRATION_OSCLASS_2_DOCKER.git
        
3. Change to the folder directory.

        $ cd MIGRATION_OSCLASS_2_DOCKER

4. Open the  in `common.env` file in your favorite editor.

        $ vi common.env 


5. Set the `DOMAIN` and `MYSQL_ROOT_PASSWORD` to contain your domain and password.

         DOMAIN=yourdomain.org
         MYSQL_ROOT_PASSWORD=yourpassword

6. Save and close the file.

7. Build containers.

        $ docker-compose build

8. Run containers.

        $ docker-compose up

9. Launch the `http://your_ip_docker_machine` URL in your favorite browses.

    If you don't know it, you can get your `ip` with the following command:

        $ docker-machine ip
     

## Import data

1. Create a `tar.gz` file of your current Osclass server.

2. Replace the `backup_osclass.tar.gz` in the `migration` folder.
     
3. Create a dump file of your Osclass database.

4. Replace the `backup.mysql.sql` in the `migration` folder.
     
5. Copy the `migration` folder into the Osclass container.

        $ tar -cf - migration | docker exec -i <NAME_OSCLASS_CONTAINER>  /bin/tar -C / -xf -

6. Run the `osclass_init.sh` script to import the files and database data into the volume containers.

        $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_init.sh

## Backups

Generate a backup of our osclass files and database.

    $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_backup.sh

## Others

There is an `INSTALL.sh` file if you want to launch without `docker-compose`.